### ISSUE TITLE
Computing contributions w.r.t. logits rather than final activations

### ISSUE THREAD

**Issue by AvantiShri** [pytorch/captum#91]
Often, in practice, we wish to compute the contributions w.r.t. the logits of the final sigmoid/softmax, rather than w.r.t. the final network output itself. This is to avoid artifacts that can be caused by the saturating nature of the sigmoid/softmax, and comes into play when comparing attributions between examples. It is particularly relevant if gradient*input is used as an attribution method, because for examples with very confident predictions, the sigmoid/softmax outputs tend to saturate and the gradients will approach zero. I'm wondering if it may be worth mentioning this in the documentation - in the current "getting started", the toy model has a sigmoid output:

<img width="398" alt="Screenshot 2019-10-08 at 2 19 52 PM" src="https://user-images.githubusercontent.com/2302598/66434147-bcb89f80-e9d6-11e9-9cb1-981d1553664d.png">

I'm concerned that a naive user may try to compare the magnitudes of attributions across different examples without realizing that, for sigmoid/softmax outputs, it may be worth removing the final nonlinearity before doing such a comparison. We discuss this in [Section 3.6](https://arxiv.org/pdf/1704.02685.pdf) of the deeplift paper. Ideally there would be an option in Captum to ignore the final nonlinearity, but I realize it may not be trivial to add that option. Sorry if this is already addressed and I missed it.

**Comment by orionr** [comment]
@NarineK, is this finished so we can close it or is there some additional feature request here. Thanks all!

**Comment by gabrieltseng** [comment]
I am happy to give this a shot!

**Comment by NarineK** [comment]
Thank you so much, @gabrieltseng ! Of course, go ahead! 

One thing to note is that, I'm expanding DeepLIFT to support Layer and Neuron and there might be changes also in the current implementation of DeepLift because it has some issues with models that are wrapped with `DataParallel`, I'll submit the PR soon but you can start working on this feature.

**Comment by NarineK** [comment]
Merged with: #142
Closing ...
